refactor: reorganize cabal.project files

andreabedini opened the pull request:
- Improve the consistency across the stage cabal.project files by using the same ordering and delimiter comments
- "package-dbs: clear, global" is the default so it's removed

andreabedini committed:
refactor: reorganize cabal.project files

- Improve the consistency across the stage cabal.project files by using the same ordering and delimiter comments
- "package-dbs: clear, global" is the default so it's removed

angerman reviewed (approved):
Looks like an improvement. I'm a bit concerned it might mess with #75?

hasufell commented:
Yes... this is a refactoring and not a feature, so it should be lower in priority than any existing feature PR.

andreabedini committed:
fix handling of index-state

angerman committed:
fix: add missing ghc-platform to cabal.project.stage1

The refactoring in 'reorganize cabal.project files' accidentally removed
libraries/ghc-platform from cabal.project.stage1. This package is required
because ghc-boot depends on ghc-platform >= 0.1.

Without this fix, all CI builds fail with:
  Error: [Cabal-7107]
  Could not resolve dependencies:
  unknown package: host:ghc-platform (dependency of host:ghc-boot)

angerman commented:
Pretty much all of the builds now passed.
